### ISSUE TITLE
[improve] EET — experience-driven early termination for cost-efficient SE agents (arXiv:2601.05777)

### ISSUE THREAD

**Issue by cetej** [cetej/STOPA#89]
## EET — Experience-Driven Early Termination for Cost-Efficient SE Agents

**Source:** arxiv-daily-digest 2026-04-30
**Type:** research
**Relevance score:** 7 (orchestration, budget-guards, circuit-breakers)
**Actionability:** immediate

### Description
EET (arXiv:2601.05777) predicts when to stop agent loops early based on past execution trajectories — achieving cost-efficient SE agents without quality loss. Directly applicable to STOPA budget tier guards and circuit breakers: instead of fixed 3-fix escalation rules, use trajectory-based predictor to decide early termination.

### Suggested Action
1. Read arXiv:2601.05777 — extract the trajectory feature set used for early-termination prediction
2. Map features to STOPA outcomes/*.md fields (iterations, score_start, score_end, exit_reason)
3. Prototype: after N=3 iterations in /autoloop, query outcomes for similar task_class + complexity → predict if additional iterations converge or waste budget
4. If effective: wire into core-invariants as P8 (trajectory-informed circuit breaker)

### Context
- Matched domains: orchestration, multi-agent
- Matched goals: self-improving harness maturation
- Matched pain_points: circuit breakers currently rule-based (3-fix limit), not trajectory-informed

---
*Auto-generated by STOPA /improve sweep 2026-05-09*

**Comment by cetej** [comment]
## Phase 0 gate: deferred — insufficient training data

**Concrete count:** 13 outcome records total (`ls .claude/memory/outcomes/*.md | wc -l` = 13, archive empty).

**Threshold:** ≥50 records required before EET prototype is meaningful.

### Why defer

A trajectory predictor (EET, arXiv:2601.05777) needs a training corpus to learn "stop vs continue" decision boundaries. With n=13:
- Backtest signal indistinguishable from noise (wide CIs)
- Risk overfitting to current task mix (mostly koder preklad/wave runs, 2 self-evolve)
- No statistical basis to override rule-based P7 circuit breaker

Same anti-pattern that closed #91 (paper claim ≠ reality) and #88 (signal redundant with existing). Phase 0 data check before paper read prevents both.

### Existing safety net

`core-invariants.md` P7 (3-fix escalation with error classification + Reflexion) remains binding and is sufficient at current data scale.

### Auto-reopen criterion

Reopen when outcomes accumulate to ≥50:

```bash
ls .claude/memory/outcomes/*.md | wc -l  # ≥ 50 → reopen
```

At current accumulation rate (~13 records / ~30 days = 0.43/day), threshold reached around 2026-08-15 absent acceleration.

### Learning captured

[`.claude/memory/learnings/2026-05-09-eet-evaluation.md`](.claude/memory/learnings/2026-05-09-eet-evaluation.md) — generalized data-prerequisite gate applicable to /self-evolve optstate, HERA replay-queue, agent-accountability predictors.
